frontend/src/reducers/credentials.js

```diff
@@ -0,0 +1,78 @@
+import { createSlice } from '@reduxjs/toolkit'
+import { batch } from 'react-redux'
+import { API_URL } from '../reusables/urls'
+
+const credentials = createSlice({
+    name: 'credentials',
+    initialState: {
+        username: null,
+        accessToken: null,
+        error: null,
+        secret: ''
```

Comment:
You need to store your data in the local storage to avoid logging out your user after refreshing the page.